spec/components/javascript_required_component_spec.rb

```diff
@@ -25,9 +25,7 @@
   end
 
   it 'loads css resource for setting session key in JavaScript-disabled environments' do
-    expect(rendered).to have_css('noscript link') do |node|
-      node[:href] == no_js_detect_css_path
-    end
+    expect(rendered).to have_css("noscript link[href='#{no_js_detect_css_path}']")
```

Comment:
This seemingly unrelated change is because I was trying to incorporate [the `optional_block_filter` feature of `have_field`](https://www.rubydoc.info/gems/capybara/Capybara%2FRSpecMatchers:have_field) in the spec that I did add, something like...

```rb
expect(rendered).to have_field('username', type: :text, with: user.email, visible: false) do |node|
  node[:autocomplete] == 'username'
end
```

But I discovered that apparently it doesn't actually filter at all? So the `node[:href]` check wasn't doing anything here previously (you could change `==` to `!=` and it would still pass).